### PR TITLE
Serial scheduling

### PR DESCRIPTION
**Pull request by weekface**
This pr fixes: #264, make sure only one pod can be scheduled at the same time.

### CODE REVIEW

File: pkg/scheduler/predicates/ha.go [hunk 1]
```diff
@@ -127,13 +142,69 @@ func (h *ha) Filter(instanceName string, pod *apiv1.Pod, nodes []apiv1.Node) ([]
...
+	if err != nil {
+		return nil, nil, err
+	}
+
+	currentPVCName := pvcName(component, podName)
+	var currentPVC *apiv1.PersistentVolumeClaim
+	var schedulingPVC *apiv1.PersistentVolumeClaim
+	for i := 0; i < len(pvcList.Items); i++ {
+		pvc := pvcList.Items[i]
+		if pvc.GetName() == currentPVCName && currentPVC == nil {
```
Comment (onlymellb) [inline pkg/scheduler/predicates/ha.go]: Why do you want to add the condition `currentPVC == nil`, will there be multiple PVCs with the same name?

Comment (weekface) [reply pkg/scheduler/predicates/ha.go]: Removed.

File: pkg/scheduler/predicates/ha.go [hunk 2]
```diff
@@ -127,13 +142,69 @@ func (h *ha) Filter(instanceName string, pod *apiv1.Pod, nodes []apiv1.Node) ([]
...
+	podName := pod.GetName()
+	pvcList, err := h.pvcListFn(ns, instanceName, component)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	currentPVCName := pvcName(component, podName)
+	var currentPVC *apiv1.PersistentVolumeClaim
+	var schedulingPVC *apiv1.PersistentVolumeClaim
+	for i := 0; i < len(pvcList.Items); i++ {
```
Comment (onlymellb) [inline pkg/scheduler/predicates/ha.go]:
```suggestion
	for _, pvc := range pvcList.Items {
```

Comment (weekface) [reply pkg/scheduler/predicates/ha.go]: Use this style to the golang [common mistakes](https://github.com/golang/go/wiki/CommonMistakes)

Comment (onlymellb) [reply pkg/scheduler/predicates/ha.go]: There is not use closures running as goroutines，If there is such a scene, then the way to use range can also avoid this problem.

Comment (weekface) [reply pkg/scheduler/predicates/ha.go]: `currentPVC ` and `schedulingPVC ` are pointers,  the `pvc` in the for range block is a variable,    if we use the for range style, then these two variables will get the same address.

Comment (weekface) [reply pkg/scheduler/predicates/ha.go]: For example:
```
package main

import "fmt"

func main() {
	arr := []string{"a", "b", "c"}

	var pa *string
	var pb *string
	for _, item := range arr {
		if item == "a" {
			pa = &item
		}
		if item == "b" {
			pb = &item
		}
	}
	fmt.Println(pa)
	fmt.Println(pb)
	fmt.Println(pa == pb)

	var ppa *string
	var ppb *string
	for i := 0; i < len(arr); i++ {
		if arr[i] == "a" {
			ppa = &arr[i]
		}
		if arr[i] == "b" {
			ppb = &arr[i]
		}
	}
	fmt.Println(ppa)
	fmt.Println(ppb)
	fmt.Println(ppa == ppb)
}
```

Outputs:
```
0xc00005e1c0
0xc00005e1c0
true
0xc000060180
0xc000060190
false
```

Comment (onlymellb) [reply pkg/scheduler/predicates/ha.go]: You can create a new variable to fix this problem. such as follow:
```
for i := 0; i < len(pvcList.Items); i++ { 
    pvc := pvcList.Items[i]
    ...
}
------
for _, pvc := range pvcList.Items {
    pvc := pvc
 ...
}
```

Comment (weekface) [reply pkg/scheduler/predicates/ha.go]: Style `for i:=0; i<n; i++` is equivalent to the `for range` style, but using `for i:=0; i<n; i++` is more clear at this situation.

**Pull request merged**